go_modules/lib/dependabot/go_modules/file_parser.rb

```diff
@@ -44,6 +56,36 @@ def ecosystem
 
       private
 
+      sig { void }
+      def set_go_environment_variables
+        # If a go.env file is present, we set the GOENV environment variable
+        # so that the Go toolchain can use it, this is useful for setting
+        # GOPRIVATE and GONOSUMDB for example.
+        if go_env
+          T.must(go_env)
+          File.write(T.must(go_env).name, T.must(go_env).content)
+          go_env_path = Pathname.new(T.must(go_env).name).realpath.to_s
+          ENV["GOENV"] = go_env_path
+        end
```

Comment:
In theory you could always set `GOENV` to the repo without checking the file exists. The Go CLI will not error if it doesn't find a go.env there. Tried it locally seems to be ok with it.

Reply:
Hmm I think I prefer safeguarding since it's cheap, just in case the behavior ever changes